### ISSUE TITLE
Assorted small cleanups: `load_config` docstring, enc/dec dim coupling, `torch.roll` dims

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#35]
Small things found while reading, none of which change results today.

**1. `load_config` docstring is wrong** (`rhythmic_relationships/model_utils.py:13-20`)

```python
def load_config(filepath):
    """Loads a model config and adds derived values"""
```

It only casts `lr` to `float`. The derived values (`vocab_size`, `context_len`, `pad_ix`) are computed ad hoc in each `train()`, differently — which is how #8 ended up setting `vocab_size` to a context length. Either move that derivation into `load_config` or fix the docstring.

**2. `hits_encdec` silently requires `enc_n_embed == dec_n_embed`**

`models/hits_encdec.py` passes the encoder output as `context` to the decoder without a `cross_attn_dim`, so cross-attention only works while the two embedding widths match. Every config satisfies this (144/144), so it works — but nothing says so. An assert in `__init__` would turn a confusing shape error into a clear one.

**3. `torch.roll(yb, 1)` without `dims` is correct but fragile**

```python
# scripts/modeling/hits_encdec/utils.py:5-6
yb_shifted = torch.roll(yb, 1)
yb_shifted[:, 0] = torch.zeros((yb.shape[0],))
```

With `dims=None`, `torch.roll` flattens, rolls, and reshapes — so row `b` column 0 receives row `b-1`'s last token. That element is immediately overwritten with the start token, which makes the result **identical** to `torch.roll(yb, 1, dims=1)`:

```
flatten-roll == dim-1 roll after fixing col 0: True
```

So this is not a bug. But it's only correct by accident of the next line, and it reads as a bug. Pass `dims=1` explicitly.

**4. `save_bento_model` is unreferenced.** `model_utils.py:56` — no training script calls it, and `bento: False` in every config. It's the only reason `bentoml` is a dependency (#34).
